content/en/docs/components/pipelines/sdk/component-development.md

```diff
@@ -92,296 +100,459 @@ def do_work(input1_file, output1_file, param1):
+
+<a name="component-spec"></a>
+## Creating a component specification
+
+To create a component from your containerized program, you must create a
+component specification that defines the component's interface and
+implementation. The following sections provide an overview of how to create a
+component specification by demonstrating how to define the component's
+implementation, interface, and metadata.
+
+To learn more about defining a component specification, see the 
+[component specification reference guide][component-spec].
+
+[component-spec]: /docs/components/pipelines/reference/component-spec/
+
+### Define your component's implementation
+
+The following example creates a component specification YAML and defines the
+component's implementation. 
+
+1.  Create a file named `component.yaml` and open it in a text editor.
+1.  Create your component's implementation section and specify the strict name
+    of your container image. The strict image name is provided when you run
+    your `build_image.sh` script.
+
+    ```yaml
+    implementation:
+      container:
+        # The strict name of a container image that you've pushed to a container registry.
+        image: gcr.io/my-org/my-image@sha256:a172..752f
+    ```
+
+1.  Define a `command` for your component's implementation. This field
+    specifies the command-line arguments that are used to run your program in
+    the container. 
+
+    ```yaml
+    implementation:
+      container:
+        image: gcr.io/my-org/my-image@sha256:a172..752f
+        # command is a list of strings (command-line arguments). 
+        # The YAML language has two syntaxes for lists and you can use either of them. 
+        # Here we use the "flow syntax" - comma-separated strings inside square brackets.
+        command: [
+          python3, 
+          # Path of the program inside the container
+          /pipelines/component/src/program.py,
+          --input1-path,
+          {inputPath: Input 1},
+          --param1, 
+          {inputValue: Parameter 1},
+          --output1-path, 
+          {outputPath: Output 1},
+        ]
+    ```
+
+    The `command` is formatted as a list of strings. Each string in the
+    `command` is a command-line argument or a placeholder. At runtime,
+    placeholders are replaced with an input or output. In the preceding
+    example, two inputs and one output path are passed into a Python script at
+    `/pipelines/component/src/program.py`.
+
+    There are three types of input/output placeholders:
+
+    *   `{inputValue: <input-name>}`:
+        This placeholder is replaced with the value of the specified input.
+        This is useful for small pieces of input data, such as numbers or small
+        strings.
+
+    *   `{inputPath: <input-name>}`:
+        This placeholder is replaced with the path to this input as a file.
+        Your component can read the content's of that input at that path during
```

Comment:
Remove apostrophe from "content's" to make it: contents

Reply:
Thanks for catching that!